integrations/oci/oci/src/main/java/io/helidon/integrations/oci/AtnStrategyInstancePrincipal.java

```diff
@@ -63,18 +64,25 @@ public Optional<AbstractAuthenticationDetailsProvider> provider() {
 
     private static LazyValue<Optional<AbstractAuthenticationDetailsProvider>> createProvider(OciConfig config) {
         return LazyValue.create(() -> {
-            if (imdsVailable(config)) {
-                return Optional.of(InstancePrincipalsAuthenticationDetailsProvider.builder().build());
+            if (imdsAvailable(config)) {
+                var builder = InstancePrincipalsAuthenticationDetailsProvider.builder()
+                        .timeoutForEachRetry((int) config.atnTimeout().toMillis());
+
+                config.imdsBaseUri()
+                        .map(URI::toString)
+                        .ifPresent(builder::metadataBaseUrl);
```

Comment:
@klustria fiy

Reply:
Can we add `config.imdsBaseUri()` in builder if it exist? `InstancePrincipalsAuthenticationDetailsProvider.builder()` has a method `.metadataBaseUrl(metadataBaseUrl)` that allows setting of this. When this is not set, this API will use the default, but if set, then it can be used in combination with ssh tunnel to run Instance Principal authentication from a laptop.

Reply:
This is exactly as the code is written right now?

```
config.imdsBaseUri() // configured IMDS URI (optional, if not configured, use default)
                        .map(URI::toString) // as string
                        .ifPresent(builder::metadataBaseUrl); // set on the InstancePrincipalAtuhenticationDetailsProvider builder
```

Reply:
Ignore my comment, you are right.